com.unity.visualeffectgraph/Editor/GraphView/Views/VFXPaste.cs

```diff
@@ -47,11 +47,52 @@ public static void PasteBlocks(VFXViewController viewController, object data, VF
             s_Instance.PasteBlocks(viewController, (data as SerializableGraph).operators, targetModelContext, targetIndex, blocksInTheSameOrder);
         }
 
+        static bool CanPasteSubgraph(VisualEffectSubgraph subgraph, string openedAssetPath)
+        {
+            var path = AssetDatabase.GetAssetPath(subgraph);
+            if (path == openedAssetPath)
+            {
+                return false;
+            }
+
+            var resource = VisualEffectResource.GetResourceAtPath(path);
+            var controller = new VFXViewController(resource);
```

Comment:
It seems really heavy to create a controller on every resources to check the paste subgraph, did you consider to "statify" this behavior ?
By the way, you can access to the list of children of a graph without creating a controller.

Reply:
Notice your change here: https://github.com/Unity-Technologies/Graphics/pull/6378/files#diff-08a12858efc2dbffc2c7848ae6458e070543e1d2453fd35a27b0d990a315dcf1R59